src/Workspaces/Core/Portable/Workspace/Workspace_Registration.cs

```diff
@@ -42,7 +43,12 @@ protected void RegisterText(SourceTextContainer textContainer)
 
         var registration = GetWorkspaceRegistration(textContainer);
         registration.SetWorkspace(this);
-        this.ScheduleTask(registration.RaiseEvents, "Workspace.RegisterText");
+
+        // Require main thread on the callback as WorkspaceRegistration.RaiseEvents invokes Workspace.WorkspaecChanges which is publicly exposed
```

Comment:
```suggestion
        // Require main thread on the callback as WorkspaceRegistration.RaiseEvents invokes Workspace.WorkspaceChanges which is publicly exposed
```

Reply:
This may want to have a reference to #32551 since this is really fishy that we're doing that. (I appreciate the comment in WorkspaceRegistration.RaiseEvents commenting we'll fix that in "preview 3".)